File: CPP/Greedy Approach/Find the Minimum Number of Fibonacci Numbers Whose Sum Is K.cpp
class Solution {
public:
    int findMinFibonacciNumbers(int k) {
        vector<int> fib = {1, 1};
        while (fib[fib.size() - 1] <= k) {
            fib.push_back(fib[fib.size() - 1] + fib[fib.size() - 2]);
        }
        int cnt = 0;
        int i = fib.size() - 1;
        while (k > 0) {
            if (fib[i] <= k) {
                k -= fib[i];
                cnt++;
            }
            i--;
        }
        return cnt;
    }
};

File: Java/Greedy Approach/Find the Minimum Number of Fibonacci Numbers Whose Sum Is K.java
class Solution {
    public int findMinFibonacciNumbers(int k) {
        List<Integer> fib = new ArrayList<Integer>();
        fib.add(1);
        fib.add(1);
        while (fib.get(fib.size() - 1) <= k) {
            fib.add(fib.get(fib.size() - 1) + fib.get(fib.size() - 2));
        }
        int cnt=0;
        int i = fib.size() - 1;
        while (k > 0) {
            if (fib.get(i) <= k) {
                k -= fib.get(i);
                cnt++;
            }
            i--;
        }
        return cnt;
    }
}
